src/OrleansServiceBus/Providers/Streams/EventHub/EventDataExtensions.cs

```diff
@@ -31,14 +33,14 @@ public static string GetStreamNamespaceProperty(this EventData eventData)
         public static byte[] SerializeProperties(this IDictionary<string, object> properties)
         {
             var writeStream = new BinaryTokenStreamWriter();
-            SerializationManager.Serialize(properties, writeStream);
+            SerializationManager.Serialize(properties.Where(kvp => !SkipProperties.Contains(kvp.Key)).ToList(), writeStream);
```

Comment:
This is an optimization.  Serialized dictionaries expand to more data than lists.  The skip and to list are mainly intended for optimization purposes.